precompile/modules/registerer.go

```diff
@@ -0,0 +1,50 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package modules
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	// registeredModules is a list of Module to preserve order
+	// for deterministic iteration
+	registeredModules = make([]Module, 0)
+)
+
+// RegisterModule registers a stateful precompile module
+func RegisterModule(stm Module) error {
+	address := stm.Address
+
+	for _, registeredModule := range registeredModules {
+		if registeredModule.Address == address {
+			return fmt.Errorf("address %s already used by a stateful precompile", address)
```

Comment:
Not sure this error case is tested

Reply:
Fixed https://github.com/Kava-Labs/go-ethereum/pull/10/commits/671ae7bed3d5d22a89e052acb053c9968982c921